### DISCUSSION TITLE
Extract or Match "101 Switching Protocols" for WebSockets

### DISCUSSION THREAD

**Discussion by Sim4n6** [projectdiscovery/nuclei-templates#4299] [Q&A] [answered]
It seems that no HTTP response is recorded in debug mode with the status code `101 Switching Protocols` in case of a request with the following headers: 

```
Connection: Upgrade
Sec-WebSocket-Key: U2NqiNSUA==
Sec-WebSocket-Version: 13
Upgrade: websocket
```
So, I was wondering is there a way to catch (extract or match) a **101** status code using nuclei scanner, please.

**Comment by olearycrew** [comment]
Could you try this with a `raw` HTTP query perhaps?

**Reply by Sim4n6** [reply]
The issue was asked in 2022. Thank you

**Reply by Sim4n6** [reply]
Does nuclei support WebSocket protocol , please ?

**Comment by ehsandeep** [comment, marked as answer]
Yes, now you can @Sim4n6, I've also added detection template in nuclei-templates project - https://github.com/projectdiscovery/nuclei-templates/pull/7069
